microprofile/WFLY-12342_server_readiness_probes.adoc

```diff
@@ -0,0 +1,95 @@
+= [WFLY-12342] Integrate server probes in MP Health readiness check
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,health,openshift
+
+== Overview
+
+MicroProfile Health 2.0 introduced a readiness health check.
+WFLY-12228 added support for MicroProfile Health 2.0 without enhancing the readiness health check.
+
+This means that the readiness probe always return UP by default (as long as the microprofile-health-smallrye is installed).
+
+This RFE enhances it with 3 probes (that are part of the https://github.com/jboss-container-images/jboss-eap-modules/blob/251f422c97e1bcd3625f57295bc79973193482a6/os-eap-probes/2.0/added/probes/probe/eap/dmr.py#L23[EAP readiness probe script]):
+
+* `server-status` - returns UP when the server-state is `running`
+* `boot-errors` - return UP if there are no boot-errors
+* `deployment-status` - returns UP if status for all deployments is OK
+
+Once this RFE is done, we could simplify the EAP readiness probe script by replacing its various DMR checks by a single check to `/subsystem=microprofile-health-smallrye:check-ready` (or a HTTP call to `http://:9990/health/ready`).
```

Comment:
@jmesnil What about domain mode? Is it supported? Or this is just meant for standalone mode?

When playing locally, I don't see `microprofile-health-smallrye` subsystem by default.

Reply:
MicroProfile Health is not supported in domain mode, I'll update the dev analysis to make that explicit